### DISCUSSION TITLE
Publishing download links in the check details instead of posting a comment

### DISCUSSION THREAD

**Discussion by ghost** [community/community#25585] [Actions]
<p>Hi, we just built a workflow to build a preview version in a pull-request, and push this build to our download server: <a href="https://github.com/deltachat/deltachat-desktop/blob/master/.github/workflows/build-preview.yml" rel="nofollow noopener">https://github.com/deltachat/deltachat-desktop/blob/master/.github/workflows/build-preview.yml</a></p>
<p>Part of the workflow is that we want to provide the download links to the contributors directly. Right now we do this by posting a comment to the PR, but it’s very noisy to have a comment for every commit. Instead we’d like to display the download links in the checks overview below, e.g. where it says “Details”.</p>
<p>Is there a way to do this conveniently? I couldn’t find anything in the documentation, neither could I find an Action or Job which does it already, there is only the one for posting comments.</p>

**Comment by arnested** [comment]
<p>You can post it to GitHubs API yourself.</p>
<p>We have som project where we do something along this (shell script):</p>
<pre><code>GITHUB_API_URL="https://api.github.com/repos/${PROJECT_USERNAME}/${PROJECT_REPONAME}/statuses/${SHA1}"

STATE=success

STATUS_DATA="{\"state\": \"${STATE}\", \"description\": \"Report\", \"context\": \"My status\", \"target_url\": \"${ARTIFACT_URL}\"}"

curl --fail --silent -X POST --user ":${GITHUB_TOKEN}" "${GITHUB_API_URL}" --data "${STATUS_DATA}"
</code></pre>
<p>(ok, we do that from CircleCI but I think you should be able to do it from an action as well.</p>

**Comment by ghost** [comment]
<p>cool, will try it out!</p>

**Comment by ghost** [comment]
<p>Hm, I’m playing around with it, but the API doesn’t seem to do anything. Neither does it give me a proper error message. Where would the message pop up?</p>
<p>Do I need to use a user with write access to the repository, or is it okay to use the Github actions bot for this? Or is there something else I did wrong?</p>
<p><a href="https://github.com/deltachat/deltachat-desktop/pull/1116/files#diff-e897be051354af89d81ad484418359d5R74" rel="nofollow noopener">https://github.com/deltachat/deltachat-desktop/pull/1116/files#diff-e897be051354af89d81ad484418359d5R74</a></p>
<p>The request itself looks fine to me in the logs: <a href="https://github.com/deltachat/deltachat-desktop/pull/1116/checks?check_run_id=291090229" rel="nofollow noopener">https://github.com/deltachat/deltachat-desktop/pull/1116/checks?check_run_id=291090229</a></p>

**Comment by arnested** [comment]
<p>Weird. I think somethings wrong with SHA, but cannot figure out why.</p>
<p>Your action posts the status to the SHA c239d9ac7d5b5a47d2fc28892c55cdd226facd2c but I can’t find that SHA in repos / in the PR. From the PR it looks like the SHA should be ae82c8b4a7b9f74672bf5881b0095141f1aac700.</p>

**Comment by ghost** [comment]
<p>Hm, this seems to be the merge commit which would be generated when the PR would be merged after this commit: <a href="https://github.com/deltachat/deltachat-desktop/tree/c239d9ac7d5b5a47d2fc28892c55cdd226facd2c" rel="nofollow noopener">https://github.com/deltachat/deltachat-desktop/tree/c239d9ac7d5b5a47d2fc28892c55cdd226facd2c</a></p>
<p>I read something in this direction in some thread. I guess I need that commit ID so the build job can be referenced correctly?</p>

**Comment by arnested** [comment]
<p>Could it be github.ref instaed of github.sha?</p>
<p>I had a debug dump of the github context somewhere but can’t find it …</p>
<p>Nah… It <em>should</em> be github.sha  … </p>
<p>I made a test over at <a href="https://github.com/arnested/playground/pull/4" rel="nofollow noopener">https://github.com/arnested/playground/pull/4</a> and github.sha  corresponds to the commit in PR … weird</p>

**Comment by arnested** [comment, marked as answer]
<p>Posting a status worked for me in: <a href="https://github.com/arnested/playground/pull/5" rel="nofollow noopener">https://github.com/arnested/playground/pull/5</a></p>
<p>Still no clue on what’s different in your example…</p>

**Comment by ghost** [comment]
<p>Ah, the difference is that you are using a push event, not a pull-request event. push events don’t create a merge commit, but use just their own commit’s SHA. PR events create a merge commit and use that SHA.</p>
<p>The best solution would probably be if the API would answer to the merge commit’s SHA. Do you know where I could open an issue for that?</p>

**Comment by ghost** [comment]
<p>I got it! I could get the SHA from the PR event: <a href="https://github.com/deltachat/deltachat-desktop/pull/1116/files" rel="nofollow noopener">https://github.com/deltachat/deltachat-desktop/pull/1116/files</a></p>
<p>Thanks so much for taking your time to debug this with me! I learned a lot.</p>

**Comment by arnested** [comment]
<p>Cool. I didn’t notice yours was triggered on a pull_request event.</p>
<p>I dived a bit more into it. It appears you can the right SHA in a pull request from ${{ github.event.after }} (can also be used on push events).${{ github.event.head_commit.id }}also looks like a possibility.</p>
<p>I made a new PR for testing this: <a href="https://github.com/arnested/playground/pull/6" rel="nofollow noopener">https://github.com/arnested/playground/pull/6</a></p>
<p>There is two workflows: one triggering on push and one on pull_request. Both dumps the the context for debugging and both post a status.</p>

**Comment by ghost** [comment]
<p>Uh, great. I will refactor my code I guess, that is way more beautiful.</p>

**Comment by ghost** [comment]
<p>Note for later reference; all three solutions have the problem that they can’t get the commit sha for the first commit in a PR. I tried all of them:</p>
<ul>
<li><a href="https://github.com/deltachat/deltachat-pages/pull/202" rel="nofollow noopener">https://github.com/deltachat/deltachat-pages/pull/202</a></li>
<li><a href="https://github.com/deltachat/deltachat-pages/pull/203" rel="nofollow noopener">https://github.com/deltachat/deltachat-pages/pull/203</a></li>
<li><a href="https://github.com/deltachat/deltachat-pages/pull/204" rel="nofollow noopener">https://github.com/deltachat/deltachat-pages/pull/204</a></li>
</ul>
<p>If someone finds a solution in the future, let me know :wink:</p>
